### PR TITLE
[Danger] Initial Danger Setup

### PR DESCRIPTION
**Pull request by anayini**

### CODE REVIEW

File: Dangerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# Make it more obvious that a PR is a work in progress and shouldn't be merged yet
+warn("PR is classed as Work in Progress") if github.pr_title.include? "[WIP]"
+
+# Warn when there is a big PR
+warn("Big PR") if git.lines_of_code > 500
+
+# Milestones are required to track what's included in each release
+if has_source_changes && not_declared_trivial
+  has_milestone = !github.pr_json['milestone'].nil?
+  warn('All pull requests should have a milestone attached, unless marked *#trivial*.', sticky: false) unless has_milestone
```
Comment (jessesquires) [inline Dangerfile]: how do you feel about removing `not_declared_trivial` ? i feel like we don't use it that much. and really -- it helps if everything alwyas has a milestone.

**Comment by jessesquires** [comment]
I think we should add:

change log warning:

```ruby
# Changelog entries are required for changes to library files
no_changelog_entry = !git.modified_files.include?("CHANGELOG.md")
if has_source_changes && no_changelog_entry && git.lines_of_code > 10
  fail("Source code changes (in APIs or behaviors) should have an entry in CHANGELOG.md.")
end
```

docs warning:

```ruby
# Docs are regenerated when releasing
has_doc_changes = !git.modified_files.grep(/docs\//).empty?
has_doc_gen_title = github.pr_title.include? "#docgen"
if has_doc_changes && !has_doc_gen_title
  fail("Docs are regenerated when creating new releases.")
  message("Docs are generated by using [Jazzy](https://github.com/realm/jazzy). If you want to contribute, please update the [markdown guides](https://github.com/plangrid/ReactiveLists/tree/master/Guides) or doc comments.")
end
```

borrowed from ListKit -- we often had people trying to edit the docs `.html` directly

File: Dangerfile (hunk 1 above)
Comment (Ben-G) [reply Dangerfile]: Would make sense to me. I think milestones are more important in open projects as we don't have any other tool for tracking releases.

Comment (anayini) [reply Dangerfile]: Yeah can remove that.

**Comment by anayini** [comment]
Cool will do - deleted it since didn't know our CHANGELOG situation.

File: Dangerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# Make it more obvious that a PR is a work in progress and shouldn't be merged yet
+warn("PR is classed as Work in Progress") if github.pr_title.include? "[WIP]"
+
+# Warn when there is a big PR
+warn("Big PR") if git.lines_of_code > 500
+
+# Milestones are required to track what's included in each release
+if has_source_changes
+  has_milestone = !github.pr_json['milestone'].nil?
```
Comment (jessesquires) [inline Dangerfile]: shouldn't this be `if has_source_changes && !github.pr_json['milestone'].nil?` ?

File: Dangerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  warn('All pull requests should have a milestone attached, unless marked *#trivial*.', sticky: false) unless has_milestone
```
Comment (jessesquires) [inline Dangerfile]: should change warning to "All pull requests should have a milestone attached."

File: Dangerfile [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+if has_source_changes && no_changelog_entry && git.lines_of_code > 10
+  fail("Source code changes (in APIs or behaviors) should have an entry in CHANGELOG.md.")
+end
+
+# Docs are regenerated when releasing
+has_doc_changes = !git.modified_files.grep(/docs\//).empty?
+has_doc_gen_title = github.pr_title.include? "#docgen"
+if has_doc_changes && !has_doc_gen_title
+  fail("Docs are regenerated when creating new releases.")
+  message("Docs are generated by using [Jazzy](https://github.com/realm/jazzy). If you want to contribute, please update [markdown guides](https://github.com/Instagram/IGListKit/tree/master/Guides)")
```
Comment (jessesquires) [inline Dangerfile]: lol copy 🍝 from listkit

should be:

`message("Docs are generated by using [Jazzy](https://github.com/realm/jazzy). If you want to contribute, please update the [markdown guides](https://github.com/plangrid/ReactiveLists/tree/master/Guides) or doc comments.")`

Comment (anayini) [reply Dangerfile]: 😅

**Comment by PlanBot-iOS** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: No such file or directory @ rb_sysopen - .swiftlint.yml
```
 #  from Dangerfile:30
 #  -------------------------------------------
 #  swiftlint.config_file = '.swiftlint.yml'
 >  swiftlint.lint_files(inline_mode: true)
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by anayini** [comment]
@jessesquires - can we land this?  I think I got a build failure because .swiftlint.yml wasn't merged at the time, but that would make it seem like everything else worked.

**Comment by jessesquires** [comment]
Looks like first build failed on `pod lint` (expected)

Last one has passed 🎉 and danger ran successfully
https://travis-ci.org/plangrid/ReactiveLists/jobs/332573630

(i think danger didn't comment because not all builds finished yet)

Let's land, then open a "test PR" (to keep open for testing danger, since we're merging faster than travis)

**Comment by anayini** [comment]
Cool

**Pull request merged**